src/ios/CDVOrientation.m

```diff
@@ -56,31 +56,64 @@ -(void)screenOrientation:(CDVInvokedUrlCommand *)command
+                    if (@available(iOS 16.0, *)) {
+                        value16 = [[UIWindowSceneGeometryPreferencesIOS alloc] initWithInterfaceOrientations:UIInterfaceOrientationMaskLandscapeLeft];
+                    } else {
+                        value = [NSNumber numberWithInt:UIInterfaceOrientationLandscapeLeft];
+                    }
                 } else if (orientationMask == 4){
-                    value = [NSNumber numberWithInt:UIInterfaceOrientationLandscapeRight];
+                    if (@available(iOS 16.0, *)) {
+                        value16 = [[UIWindowSceneGeometryPreferencesIOS alloc] initWithInterfaceOrientations:UIInterfaceOrientationMaskLandscapeRight];
+                    } else {
+                        value = [NSNumber numberWithInt:UIInterfaceOrientationLandscapeRight];
+                    }
                 } else if (orientationMask == 1 || (orientationMask == 3 && !UIInterfaceOrientationIsPortrait(deviceOrientation))) {
-                    value = [NSNumber numberWithInt:UIInterfaceOrientationPortrait];
+                    if (@available(iOS 16.0, *)) {
+                        value16 = [[UIWindowSceneGeometryPreferencesIOS alloc] initWithInterfaceOrientations:UIInterfaceOrientationMaskPortrait];
+                    } else {
+                        value = [NSNumber numberWithInt:UIInterfaceOrientationPortrait];
+                    }
                 } else if (orientationMask == 2) {
-                    value = [NSNumber numberWithInt:UIInterfaceOrientationPortraitUpsideDown];
+                    if (@available(iOS 16.0, *)) {
+                        value16 = [[UIWindowSceneGeometryPreferencesIOS alloc] initWithInterfaceOrientations:UIInterfaceOrientationMaskPortraitUpsideDown];
+                    } else {
+                        value = [NSNumber numberWithInt:UIInterfaceOrientationPortraitUpsideDown];
+                    }
                 }
             } else {
                 if (_lastOrientation != UIInterfaceOrientationUnknown) {
                     [[UIDevice currentDevice] setValue:[NSNumber numberWithInt:_lastOrientation] forKey:@"orientation"];
                     ((void (*)(CDVViewController*, SEL, NSMutableArray*))objc_msgSend)(vc,selector,result);
-                    [UINavigationController attemptRotationToDeviceOrientation];
+                    if (@available(iOS 16.0, *)) {
+                        [self.viewController setNeedsUpdateOfSupportedInterfaceOrientations];
+                    }
+                    else {
+                        [UINavigationController attemptRotationToDeviceOrientation];
+                    }
+
+                    // [UINavigationController attemptRotationToDeviceOrientation];
                 }
             }
-            if (value != nil) {
+            if (value != nil || value16 != nil) {
                 _isLocked = true;
-                [[UIDevice currentDevice] setValue:value forKey:@"orientation"];
```

Comment:
Hi, it seems that your fix is not working.
I tried on my app, and when I use your code I have this error:
_[Orientation] BUG IN CLIENT OF UIKIT: Setting UIDevice.orientation is not supported. Please use UIWindowScene.requestGeometryUpdate(_:)_

I'm not fluent to this langage but I guess that the problem is around there:
[[UIDevice currentDevice] setValue:value forKey:@"orientation"]; will not work with iOs16.

Thanks :)

Reply:
Hi, yes i'm using iphone 16, version 16.0 (20A362) on iPhone8.
The code patch in my app seems ok too: cordova / ionic / angular.
The context: my app is locked in portrait mode, until i need to rotate the screen by code to landscape mode to display a chart.
Do you need more informations? Hope it can help.

Reply:
how are you adding the plugin? maybe remove the plugin:
`ionic cordova plugin remove cordova-plugin-screen-orientation`
`ionic cordova plugin add https://github.com/oofaish/cordova-plugin-screen-orientation.git#ios-16`

if you already have the right version, then I suggest putting a debuger in xcode in `src/ios/CDVOrientation.m`  and just stepping through to see where it goes wrong.

Reply:
Hi, finally, we checked our CI and the way we build our app, one thing was missing: rm -rf plugins so the ipa was based on the old plugin, not on the patch.
Everything is ok, thanks for your job, it works perfectly, you were right. :)

Reply:
I have tried adding the plugin as you described (i.e. remove and then add the iOS 16 version) but when trying to build I am getting various errors such as the following, any idea what could be causing this?

`use of undeclared identifier 'UIWindowSceneGeometryPreferencesIOS'`

Reply:
Which version of Xcode are you using? I don’t think the latest?

Reply:
I am building using Ionic AppFlow, the version it is using to do this looks to be 13.4.1

Reply:
If you can’t bump that up to 14 then you ll have to deprecate based on Xcode version which my code doesn’t do. I am not sure what the ups and downs of it are and not going to have time to explore I am afraid.

Reply:
No problem, I have got in touch with Ionic to ask when they will be adding Xcode 14, in the meantime I am adding an option for the user to be able to disable the screen rotation functionality as a setting in the App to stop people on iOS 16 from getting stuck in landscape mode when the app is generally supposed to run in portrait mode.

Reply:
if use xcode 13，the following is better:

        SEL supportedInterfaceSelector = NSSelectorFromString(@"setNeedsUpdateOfSupportedInterfaceOrientations");
        [self.viewController performSelector:supportedInterfaceSelector];